application/core/QuestionTypes/ArrayMultiscale/RenderArrayMultiscale.php

```diff
@@ -317,15 +318,17 @@ public function parseSubquestionsNoDropdown(&$aData)
             $aData['answertextright'] = $answertextright;
             if ($aData['shownoanswer']) {
                 if (count($aData['labelans1']) > 0) {
-                    if (empty($this->getFromSurveySession($myfname1))) {
+                    $fname1value = $this->getFromSurveySession($myfname1);
+                    if (!isset($fname1value) || $fname1value == '') {
                         #$answer .= CHECKED;
                         $aData['aSubQuestions'][$i]['myfname1_notset'] = CHECKED;
                     } else {
                         $aData['aSubQuestions'][$i]['myfname1_notset'] = "";
                     }
                 } else {
-                    if (empty($this->getFromSurveySession($myfname0))) {
-                        $answer .= CHECKED;
+                    $fname0value = $this->getFromSurveySession($myfname0);
+                    if (!isset($fname0value) || $fname0value == '') {
```

Comment:
`$fname0value === ''` (code can be 0)

Reply:
Tets is OK because $fname0value is a string (then with "0" it's OK).

But maybe best to use `===` more clear

Reply:
ping @olleharstedt ?

Microfix not needed to fix the issue, but i think it's more clear and clean.

Reply:
We will review the fix and apply something a bit more clear.

Reply:
There's also `empty()` if you wanna be explicit about looking for all possible "falsy" values in PHP.